types/src/network_address/mod.rs

```diff
@@ -958,69 +1064,16 @@ mod test {
             parse_dns_tcp(addr.as_slice()).unwrap(),
             ((IpFilter::OnlyIp6, &dns_name, 123), expected_suffix)
         );
-
-        let addr = NetworkAddress::from_str("/dns/example.com/tcp/123/memory/44").unwrap();
-        let expected_suffix: &[Protocol] = &[Protocol::Memory(44)];
-        assert_eq!(
-            parse_dns_tcp(addr.as_slice()).unwrap(),
-            ((IpFilter::Any, &dns_name, 123), expected_suffix)
-        );
-
-        let addr = NetworkAddress::from_str("/tcp/999/memory/123").unwrap();
-        assert_eq!(None, parse_dns_tcp(addr.as_slice()));
-    }
-
-    #[test]
-    fn test_parse_noise_ik() {
-        let pubkey_str = "080e287879c918794170e258bfaddd75acac5b3e350419044655e4983a487120";
-        let pubkey = x25519::PublicKey::from_encoded_string(pubkey_str).unwrap();
-        let addr = NetworkAddress::from_str(&format!("/ln-noise-ik/{}", pubkey_str)).unwrap();
-        let expected_suffix: &[Protocol] = &[];
-        assert_eq!(
-            parse_noise_ik(addr.as_slice()).unwrap(),
-            (&pubkey, expected_suffix)
-        );
-
-        let addr =
-            NetworkAddress::from_str(&format!("/ln-noise-ik/{}/tcp/999", pubkey_str)).unwrap();
-        let expected_suffix: &[Protocol] = &[Protocol::Tcp(999)];
-        assert_eq!(
-            parse_noise_ik(addr.as_slice()).unwrap(),
-            (&pubkey, expected_suffix)
-        );
```

Comment:
I would add these tests back in, and instead just add the protocols to make it a valid address.

This tests out key functionality to pull the noise-ik from the address.